### PR TITLE
TDAP061-Edit-Functionality-Using-Redux

### PR DESCRIPTION
**Pull request by amit-ib**
Implemented edit functionality through redux

### CODE REVIEW

File: src/components/todo/TodoForm.tsx [hunk 1]
```diff
@@ -16,20 +17,22 @@ const TodoForm = () => {
   const handleAdd = (e: React.FormEvent) => {
     e.preventDefault();
     if (todotext) {
-
-      dispatch(addTodo({
+      dispatch(
+        addTodo({
+          //id: uuid(),
           id: Math.random(),
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoForm.tsx:23]: This may have duplicate id

File: src/components/todo/TodoItem.tsx [hunk 2]
```diff
@@ -3,58 +3,141 @@ import { TodoModal } from "../../models";
 import { dateConverter } from "../../utils/helper";
 import { useDispatch } from "react-redux";
 import { useState } from "react";
-import { deleteTodo, markDoneTodo } from "../../redux";
+import { deleteTodo, markDoneTodo, editTodo } from "../../store";
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoItem.tsx:6]: We discussed this yesterday, to give is some meaningful prefix or suffix.
It doesn't give any idea what these are, are they reducer/action/actionType/state/constant.

File: src/components/todo/TodoItem.tsx [hunk 3]
```diff
@@ -3,58 +3,141 @@ import { TodoModal } from "../../models";
...
+    setEditMode(true);
+  };
+
+  const editFormDataChanger = (type: string, data: string | Date) => {
+    setTodo((todoData) => {
+      return {
+        ...todoData,
+        [type]: data,
+      };
+    });
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoItem.tsx:48]: again it is having deep complication to understand, can you try what we discussed yesterday.

File: src/store/todoActions.tsx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { TodoModal } from "../models";
-import { ADD_TODO, DELETE_TODO, MARK_DONE_TODO } from "./actionList";
+import { ADD_TODO, DELETE_TODO, MARK_DONE_TODO } from "./actionTypes";
```
Comment (akash-dhande-ib) [inline src/store/todoActions.tsx:2]: 👍

File: src/store/todoActions.tsx [hunk 5]
```diff
@@ -21,3 +21,10 @@ export const markDoneTodo = (data: TodoModal) => {
     payload: data,
   };
 };
+
+export const editTodo = (data: TodoModal) => {
+  return {
+    type: "EDIT_TODO",
```
Comment (akash-dhande-ib) [inline src/store/todoActions.tsx:27]: Please use constant here.

**Review by akash-dhande-ib — CHANGES_REQUESTED** [review]
Added few comments.

File: src/components/todo/TodoForm.tsx (hunk 1 above)
Comment (amit-ib) [reply src/components/todo/TodoForm.tsx]: Implemented UUID

File: src/components/todo/TodoItem.tsx (hunk 2 above)
Comment (amit-ib) [reply src/components/todo/TodoItem.tsx]: Updated

File: src/components/todo/TodoItem.tsx (hunk 3 above)
Comment (amit-ib) [reply src/components/todo/TodoItem.tsx]: Updated as discussed

File: src/store/todoActions.tsx (hunk 5 above)
Comment (amit-ib) [reply src/store/todoActions.tsx]: Updated, my mistake it was, I missed it

**Comment by amit-ib** [comment]
Did updates as per last review comments, kindly re-review

**Comment by akash-dhande-ib** [comment]
> Did updates as per last review comments, kindly re-review

I was referring to this 
image

**Review by akash-dhande-ib — APPROVED** [review]

**Pull request merged**